### ISSUE TITLE
android ndk:No such file or directory

### ISSUE THREAD

**Issue by skyjie6** [FreeRDP/FreeRDP#8671]
I want to build the library for Android on  Ubuntu 22.04 , x86_64
run 
`sudo ./scripts/android-build-freerdp.sh 
--ndk /home/dangjie/Android/Sdk/ndk/21.0.6113669 
--sdk /home/dangjie/Android/Sdk/cmake/3.22.1 
--openh264-ndk /home/dangjie/Android/Sdk/ndk/android-ndk-r15c 
--conf ./scripts/android-build-64.conf`

after that it take sometime to build the .so until the error was thrown

- [RUN] cp libcrypto.so libssl.so /home/dangjie/project/FreeRDP/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/
- [RUN] cd /home/dangjie/project/FreeRDP
- [RUN] cp -L -R /home/dangjie/project/FreeRDP/scripts/../build/openssl/include/openssl /home/dangjie/project/FreeRDP/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a/include/
- [RUN] cd /home/dangjie/project/FreeRDP
- [RUN] mkdir -p /home/dangjie/project/FreeRDP/scripts/../build/freerdp-build/arm64-v8a
- [RUN] cd /home/dangjie/project/FreeRDP/scripts/../build/freerdp-build/arm64-v8a
- [RUN] export ANDROID_NDK=/home/dangjie/Android/Sdk/ndk/21.0.6113669
- [RUN] -DANDROID_NDK=/home/dangjie/Android/Sdk/ndk/21.0.6113669 -DANDROID_NATIVE_API_LEVEL=android-21 -DCMAKE_TOOLCHAIN_FILE=/home/dangjie/Android/Sdk/ndk/21.0.6113669/build/cmake/android.toolchain.cmake -DCMAKE_BUILD_TYPE=Release -DFREERDP_EXTERNAL_PATH=/home/dangjie/project/FreeRDP/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs -DWITH_CLIENT_SDL=OFF -DCMAKE_MAKE_PROGRAM=make -DWITH_OPENH264=ON -DWITH_MEDIACODEC=OFF -DWITH_FFMPEG=ON -DWITH_SWCALE=ON -DANDROID_ABI=arm64-v8a -DCMAKE_INSTALL_PREFIX=/home/dangjie/project/FreeRDP/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a -DCMAKE_INSTALL_LIBDIR=. /home/dangjie/project/FreeRDP/scripts/..
- ./scripts/android-build-common.sh: line 85: -DANDROID_NDK=/home/dangjie/Android/Sdk/ndk/21.0.6113669: **No such file or directory**
- **[ERROR]** -DANDROID_NDK=/home/dangjie/Android/Sdk/ndk/21.0.6113669 -DANDROID_NATIVE_API_LEVEL=android-21 -DCMAKE_TOOLCHAIN_FILE=/home/dangjie/Android/Sdk/ndk/21.0.6113669/build/cmake/android.toolchain.cmake -DCMAKE_BUILD_TYPE=Release -DFREERDP_EXTERNAL_PATH=/home/dangjie/project/FreeRDP/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs -DWITH_CLIENT_SDL=OFF -DCMAKE_MAKE_PROGRAM=make -DWITH_OPENH264=ON -DWITH_MEDIACODEC=OFF -DWITH_FFMPEG=ON -DWITH_SWCALE=ON -DANDROID_ABI=arm64-v8a -DCMAKE_INSTALL_PREFIX=/home/dangjie/project/FreeRDP/scripts/../client/Android/Studio/freeRDPCore/src/main/jniLibs/arm64-v8a -DCMAKE_INSTALL_LIBDIR=. /home/dangjie/project/FreeRDP/scripts/.. retured 127

I'm sure that ndk directory is exist becasue I can cd to the file.
Also, libc6:i386 libncurses5:i386 libstdc++6:i386 -y are installed .

i have no idea to solve it, thanks

**Comment by akallabeth** [comment]
1. which branch / tag / release are you using?
2. are you sure you are working on a clean checkout? this looks like a command was split in 2 lines
3. the `NDK` should be 64 bit nowadays, so no need for `:i386` packages

**Comment by skyjie6** [comment]
> 

1.just `git clone https://github.com/FreeRDP/FreeRDP.git` ,it looks like it is master branch.
2.full command
`sudo ./scripts/android-build-freerdp.sh --ndk /home/dangjie/android-ndk-r25b --sdk /home/dangjie/Android/Sdk/cmake/3.22.1/ --openh264-ndk /home/dangjie/Android/Sdk/ndk/android-ndk-r15c/ --conf ./scripts/android-build-64.conf 
`
the reason it looks like be split 2 lines is i edit it in the issue in order to read easily.

3. the packect i386 can be ignored . just provide more info.
4. the directory can be cd to the path . and sure the read and write permision is "drwxrwxr-x"
5. try to input the path with tab key. confirming the path name is corrent.
6. try to hard code the path: DANDROID_NDK , DFREERDP_EXTERNAL_PATH, DCMAKE_INSTALL_PREFIX,DCMAKE_INSTALL_LIBDIR. but it get the same error.
7. try to change the ndk to 21.0.6113669,25.2.9519653,android-ndk-r25b. but it get the same error.

**Comment by akallabeth** [comment]
why tha hell is everybody prefixing everything with `sudo`? (sorry for that, but you´re not the first to do so, please, where is that coming from?)
`NEVER` use `sudo` unless told so for a very good reason. (e.g. you need permission to install something to system directories)

please retry without that (remove the checkout first, now you will most likely need `sudo` for that as files are now owned by `root`)

**Comment by akallabeth** [comment]
@skyjie6 ok, got it, you do not have `CMake` from the `SDK Tools` of android `SDK`

**Comment by skyjie6** [comment]
> @skyjie6 ok, got it, you do not have `CMake` from the `SDK Tools` of android `SDK`

thank you for your help. Unfortunately, it hasn't been solved yet.
i follow your suggest to clone new branchce.
1.`git clone https://github.com/FreeRDP/FreeRDP.git`
2.cd to the "FreeRDP"
3.`./scripts/android-build-freerdp.sh --ndk /home/dangjie/Android/Sdk/ndk/25.2.9519653/ --sdk /home/dangjie/Android/Sdk/cmake/ --openh264-ndk /home/dangjie/Android/Sdk/ndk/android-ndk-r15c/ --conf ./scripts/android-build-64.conf`
4.it finally get the same error.

CMake is downloading from Android Studio SDKManager. inclued  3.22.1 / 3.18.1 / 3.10.2.4988404 / 3.6.4111459.
Android SDK Location is  "/home/dangjie/Android/Sdk". so i should have CMake of Android SDK. or is it need any other else?

branche "stateable-2.0" and tag  "2.9.0" "2.8.1" will get the 404 error when first build.

**Comment by akallabeth** [comment]
I´d try the actual `SDK` path as your command line ;)
```
./scripts/android-build-freerdp.sh --ndk ~/Android/Sdk/ndk/25.2.9519653/ --sdk ~/Android/Sdk/ --openh264-ndk ~/Android/Sdk/ndk/18.1.5063045/ --conf scripts/android-build-64.conf
```

(NOTE: `openh264` is picky on which `NDK` works for which version, did succeed with the above one for the `2.3.1` version build with the config)

**Comment by skyjie6** [comment]
> I´d try the actual `SDK` path as your command line ;)
> 
> ```
> ./scripts/android-build-freerdp.sh --ndk ~/Android/Sdk/ndk/25.2.9519653/ --sdk ~/Android/Sdk/ --openh264-ndk ~/Android/Sdk/ndk/18.1.5063045/ --conf scripts/android-build-64.conf
> ```
> 
> (NOTE: `openh264` is picky on which `NDK` works for which version, did succeed with the above one for the `2.3.1` version build with the config)

okay..you are correct.
i execute the command line
`cmake -version`
and get there is no cmake.so i solved the issue after run
`sudo apt install cmake`
In conclusion, when i input the <sdk location> it will get the error **coulod not found /bin/cmake.**so i change <sdk locaition> to /sdk/cmake/. but it is incorrect, i just need to install cmake for ubuntu and set the correct sdk location.

thank you.

**Comment by akallabeth** [comment]
@skyjie6 ok, found your issue. Fixed with #8678 and #8677 for stable and master.
